src/Functions/generateUUIDv7.cpp

```diff
@@ -185,26 +179,24 @@ class FunctionGenerateUUIDv7Base : public IFunction
 
             UUIDv7Range uuid_range;
             UUIDv7 available_uuid;
-            size_t done = 0;
+            size_t processed = 0;
 
-            while (done != input_rows_count)
+            while (processed != input_rows_count)
             {
-                {
-                    Data data;
-                    uuid_range = data.reserveRange(input_rows_count - done);
-                }
+                Data data;
```

Comment:
I was rereading the diff and I've noticed this. Data holds the mutex guard, removing the scope makes performance worse as the guard will only be released on the destructor, after the for loop.